students/rhamersky/Lesson_4/mailroom_part2.py

```diff
@@ -0,0 +1,335 @@
+        # the summary chart.
+        for row in tup_donor_names:
+            for value in range(2, len(row)):
+                total += row[value]
+                donation_count += 1
+            lst_table.append([row[0], row[1], total/donation_count, total])
+            total = 0
+            donation_count = 0
+        # Reverses the list
+        for row in lst_table:
+            row.reverse()
+        # Shorts the list form largest to smallest on the first element of each row.
+        lst_table.sort(reverse=True)
+
+        index = 0
+        # Formatting the money elements and changing all int() & float() data to str() data
+        # to use for creating the chart.
+        for row in range(len(lst_table)):
+            lst_table[row][index] = "${:,.2f}".format(lst_table[row][index])  # --> Total amount for gifts
+            lst_table[row][index + 1] = "${:,.2f}".format(lst_table[row][index + 1])  # --> Avg. gift amount
+            lst_table[row][index + 2] = str(lst_table[row][index + 2])  # --> Number of gifts
+        # Reverses list
+        for row in lst_table:
+            row.reverse()
+
+        lst_headers = ["Donor_Name", "Num_Gifts", "Average_Gift", "Total_Given"]  # --> List of headers for the chart.
+        # Calculating the max length of all the elements.
+        length1 = max(len(word) for word in lst_headers) + 10  # --> Max length for the header list
+        length2 = max(len(word) for row in lst_table for word in row)  # --> Max length for the donor table
+        # Calculates the max between the header list and the donor list
+        max_length = max(length1, length2)
+        # Creates the header for the chart
+        for heading in lst_headers:
+            print(("{0:{padding}}".format(heading, padding=max_length)), end=" ")
+        print()
+        # Creates the donor rows for the chart.
+        for row in lst_table:
+            print(("{0:{padding}}".format(row[0], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[1], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[2], padding=max_length)), end=" ")
+            print(("{0:{padding}}".format(row[3], padding=max_length)), end=" ")
+            print()
+########### END OF SUB_MENU CLASS###############
+
+# Methods defined in the files class are related to opening and renaming files.
+# Modifications to the methods are needed if the files are not for a task priority list.
+
+
+class files(object):
+    @staticmethod
+    def write_email(donors):
+        '''
+        Writes emails to the donors and saves them to text files.
+        :param donors: Tuple table that stores all the donor's donation information.
+        :return:
+        '''
+        for donor_name in donors:
+            # Open text file and assign a mode to the file.
+            newFile = open(donor_name[0] + ".txt", mode="w")
```

Comment:
per pep8 snake_case for var names not camelCase

it is preferred to use `with` statement when working with files.